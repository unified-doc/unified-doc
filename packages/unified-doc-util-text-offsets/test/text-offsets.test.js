import { hast } from './fixtures';
import textOffsets from '../lib/text-offsets';

describe('textOffsets', () => {
  it('should return a new unmodified tree', () => {
    const hast = { type: 'root', children: [] };
    expect(textOffsets(hast)).not.toBe(hast);
    expect(textOffsets(hast)).toEqual(hast);
  });

  it('should apply the correct text offsets only for text nodes', () => {
    const withTextOffsets = textOffsets(hast);
    expect(withTextOffsets).not.toEqual(hast);
    expect(withTextOffsets).toMatchSnapshot();
  });
});
