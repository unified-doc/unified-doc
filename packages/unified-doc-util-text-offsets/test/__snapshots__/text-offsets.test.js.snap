// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`textOffsets should apply the correct text offsets only for text nodes 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "children": Array [
            Object {
              "data": Object {
                "textOffset": Object {
                  "end": 4,
                  "start": 0,
                },
              },
              "position": Object {
                "end": Object {
                  "column": 25,
                  "line": 1,
                  "offset": 24,
                },
                "start": Object {
                  "column": 21,
                  "line": 1,
                  "offset": 20,
                },
              },
              "type": "text",
              "value": "some",
            },
          ],
          "position": Object {
            "end": Object {
              "column": 34,
              "line": 1,
              "offset": 33,
            },
            "start": Object {
              "column": 13,
              "line": 1,
              "offset": 12,
            },
          },
          "tagName": "strong",
          "type": "element",
        },
        Object {
          "data": Object {
            "textOffset": Object {
              "end": 12,
              "start": 4,
            },
          },
          "position": Object {
            "end": Object {
              "column": 8,
              "line": 2,
              "offset": 41,
            },
            "start": Object {
              "column": 34,
              "line": 1,
              "offset": 33,
            },
          },
          "type": "text",
          "value": "
content",
        },
      ],
      "position": Object {
        "end": Object {
          "column": 21,
          "line": 2,
          "offset": 54,
        },
        "start": Object {
          "column": 1,
          "line": 1,
          "offset": 0,
        },
      },
      "tagName": "blockquote",
      "type": "element",
    },
  ],
  "position": Object {
    "end": Object {
      "column": 21,
      "line": 2,
      "offset": 54,
    },
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "type": "root",
}
`;
